use backoff / retry for dynamo put & get

yusefnapora opened the pull request:
# This adds a backoff / retry helper to the dynamo client.  It will attempt to retry up to 4 times, backing off exponentially.   So far I haven't hit the failure path when testing, so I'd like to test some more before merging this in.  Also, it's probably a good idea to add some random 'jitter' to the delay, so we can spin up a bunch of processes without them all retrying at once.

yusefnapora committed:
use backoff / retry for dynamo put & get

vyzo commented on mediachain/datastore/dynamo.py:
```diff
@@ -25,6 +28,40 @@ def get_db():
+        else:
+            return output
+
+        delay = (BACKOFF_COEFF * (2 ** attempts)) / 1000.0
```
can you make it randomized to better smooth out the load?

vyzo replied:
also, can you add an upper bound of 60sec (prior to randomizing)?

vyzo commented on mediachain/datastore/dynamo.py:
```diff
@@ -25,6 +28,40 @@ def get_db():
 
+
+BACKOFF_COEFF = 50.0
+MAX_ATTEMPTS = 4
```
make it 10 pls.

yusefnapora committed:
add random jitter, cap delay to 60s. increase retry attempts to 10

- uses the "full jitter" approach from
  https://www.awsarchitectureblog.com/2015/03/backoff.html

vyzo commented:
:+1:
